Mathlib/Analysis/Analytic/Prod.lean

```diff
@@ -0,0 +1,60 @@
+/-
+Copyright (c) 2023 Geoffrey Irving. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Geoffrey Irving
+-/
+import Mathlib.Analysis.Analytic.Composition
+import Mathlib.Analysis.Analytic.Linear
+
+/-!
+# Analytic functions on and to product spaces
+-/
+
+noncomputable section
+
+open Topology Classical BigOperators NNReal Filter ENNReal
+
+open Set Filter Asymptotics
+
+variable {𝕜 E F G H : Type*}
+
+variable [NontriviallyNormedField 𝕜] [NormedAddCommGroup E] [NormedSpace 𝕜 E] [NormedAddCommGroup F]
+  [NormedSpace 𝕜 F] [NormedAddCommGroup G] [NormedSpace 𝕜 G] [NormedAddCommGroup H]
+  [NormedSpace 𝕜 H]
+
+/-- `AnalyticAt.comp` for a curried function -/
```

Comment:
Where are the curried functions here? In this statement `h` is not curried. The curried form of `h` would be the map from `F` to functions `G → H` sending `x \in H` to the function `y ↦ h x y`. 

In nice situations (e.g. continuous functions on compacts) the continuity of `h` as a map on `F \times G` is equivalent to continuity of its curried form as a map `F \to C(G, H)`, so one can reduce continuity problems to a single variable (at a cost of allowing the target space to be large). But I'm not sure that's even defined here; the space of functions that are analytic at a given point (and arbitrary elsewhere) isn't a Banach space in any natural way, so I don't think we can define analyticity with this space as target.

Reply:
(Just to clarify, I have nothing against the content of these lemmas, I'm just advocating for a change to the docstrings.)

Reply:
Sorry, will fix: they used to be curried and called `curry_comp`, but I fixed names and definitions to not be curried but not the doctoring.

Reply:
I’m actually going to need that Banach space (the Linf version) later on to extend Hartogs theorem from 2D to ND, but I’ll build that when I use it.